java/com/globant/services/APIcontroller.java

```diff
@@ -0,0 +1,353 @@
+
+		if (request.getQuantity() != null && request.getQuantity() > 0) {
+
+			ShoppingCart cart = cartService.findByUserName(sessionID);
+			
+			Product product = productService.findByName(request.getName());
+			
+			product.setQuantity(request.getQuantity());
+			
+			if (cart == null) {
+
+				cartService.create(sessionID);
+			}
+
+			cartService.addProductToCart(product, sessionID);
+
+			return "Products added to the cart";
+
+		} else {
+			return "Invalid product amount";
+
+		}
+	}
+
+	/**
+	 * Finds a product by category
+	 * 
+	 * @param category
+	 * @return
+	 */
+	@GetMapping("/findByCategory")
+	@ApiOperation(value = "Find product by category.", nickname = "Find product by category.")
+	public String findByCategory(Model model) {
+
+		model.addAttribute("product", new Product());
+
+		return "findByCategory";
+	}
+
+	/**
+	 * Returns a list of products for a given category
+	 * 
+	 * @param category
+	 * @return
+	 */
+	@PostMapping("/findByCategory")
+	@ApiOperation(value = "Find product by category.", nickname = "Find product by category.")
+	public String findByCategory(@ModelAttribute Product request, Model model) {
+
+		List<Product> products = productService.findByCategory(request.getCategory());
+
+		if (products.isEmpty()) {
+
+			return "home";
+
+		} else {
+
+			model.addAttribute("productList", products);
+
+			model.addAttribute("product", new Product());
+
+			return "findByCategory";
+
+		}
+	}
+
+	/**
+	 * Finds a product by category
+	 * 
+	 * @param category
+	 * @return
+	 */
+	@GetMapping("/findByName")
+	@ApiOperation(value = "Find product by name.", nickname = "Find product by name.")
+	public String findByName(Model model) {
+
+		model.addAttribute("product", new Product());
+
+		return "findByName";
+	}
+
+	/**
+	 * Returns a list of products for a given category
+	 * 
+	 * @param category
+	 * @return
+	 */
+	@PostMapping("/findByName")
+	@ApiOperation(value = "Find product by name.", nickname = "Find product by name.")
+	public String findByName(@ModelAttribute Product request, Model model) {
+
+		Product product = productService.findByName(request.getName());
+
+		List<Product> products = new ArrayList<>();
+
+		products.add(product);
+
+		if (product == null) {
+
+			return "home";
+
+		} else {
+
+			model.addAttribute("productList", products);
+
+			model.addAttribute("product", new Product());
+
+			return "findByName";
+		}
+	}
+
+	/**
+	 * Calculates the total cost
+	 * 
+	 * @return
+	 */
+	@GetMapping("/buyProducts")
+	@ResponseBody
+	@ApiOperation(value = "Buy products.", nickname = "Buy products.")
+	public String buyProducts() {
+
+		ShoppingCart cart = cartService.findByUserName(sessionID);
+
+		if (cart == null) {
+
+			return "Cart not found for user name : " + sessionID;
+
+		} else {
+
+			return "The total cost is : " + cartService.payOrder(sessionID);
+		}
+
+	}
+
+	/**
+	 * Saves the current cart
+	 * 
+	 * @return
+	 */
+	@RequestMapping(value = "/saveCart")

```

Comment:
Why so many endpoints with generic mapping, you should use PostMapping, PutMapping etc.